include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -205,47 +196,37 @@ inline void enrich_assign(Operations& operations, Turns& turns,
 template <typename Operations, typename Turns>
 inline void enrich_adapt(Operations& operations, Turns& turns)
 {
-    typedef typename boost::range_value<Turns>::type turn_type;
-    typedef typename turn_type::turn_operation_type op_type;
-    typedef typename boost::range_value<Operations>::type indexed_turn_type;
-
+    // Operations is a vector of indexed_turn_operation<>
+    // If it is empty, or contains one or two items, it makes no sense
     if (operations.size() < 3)
     {
-        // If it is empty, or contains one or two turns, it makes no sense
         return;
     }
 
-    // Operations is a vector of indexed_turn_operation<>
-
-    // Last index:
-    std::size_t const x = operations.size() - 1;
     bool next_phase = false;
+    std::size_t previous_index = operations.size() - 1;
```

Comment:
Earlier it was slightly different, leading to a reproducable gcc-5 compilation error.
```
./boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp:206:54: internal compiler error: in tsubst_copy, at cp/pt.c:13040
     std::size_t const last_index = operations.size() - 1;
```

This now works for all compilers AFAICS